numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4056,9 +4118,19 @@ initialize_numeric_types(void)
 
     PyStringArrType_Type.tp_repr = stringtype_repr;
     PyStringArrType_Type.tp_str = stringtype_str;
+    if (PyString_Type.tp_as_number)
+        stringtype_as_number = *PyString_Type.tp_as_number;
+    stringtype_as_number.nb_int = int_from_string;
+    stringtype_as_number.nb_float = float_from_string;
```

Comment:
Why can't you just use `PyString_Type.nb_int` here?

Reply:
Can you just use `PyNumber_Int(PyObject_Str(self))` here instead (swapping `str` for `unicode`, and `int` for `float` as needed)`